### ISSUE TITLE
Add patches for the linux kernel (mainline/6.10)

### ISSUE THREAD

**Issue by theoparis** [aws/aws-lc#1712]
I ran into a few errors when I tried to build the linux kernel with aws-lc installed to /usr/local globally.
![image](https://github.com/user-attachments/assets/3207e0fb-882f-4961-ac26-a28cce94efaa)

I was wondering if aws-lc should add the following patches to the repository/CI since there are already other projects being patched such as cpython. These patches are partially taken from [android-mainline](https://android.googlesource.com/kernel/common/+/ab5f2e09abf7c7d8b896a231365b23e85799c3c1%5E2..ab5f2e09abf7c7d8b896a231365b23e85799c3c1/). I changed OPENSSL_IS_BORINGSSL to OPENSSL_IS_AWSLC.

```diff
diff --git a/certs/extract-cert.c b/certs/extract-cert.c
index 70e9ec89d87d..2e7d11c9bf03 100644
--- a/certs/extract-cert.c
+++ b/certs/extract-cert.c
@@ -56,6 +56,7 @@ static void display_openssl_errors(int l)
 	}
 }
 
+#ifndef OPENSSL_IS_AWSLC
 static void drain_openssl_errors(void)
 {
 	const char *file;
@@ -65,6 +66,7 @@ static void drain_openssl_errors(void)
 		return;
 	while (ERR_get_error_line(&file, &line)) {}
 }
+#endif
 
 #define ERR(cond, fmt, ...)				\
 	do {						\
@@ -99,8 +101,6 @@ int main(int argc, char **argv)
 	char *cert_src;
 	char *verbose_env;
 
-	OpenSSL_add_all_algorithms();
-	ERR_load_crypto_strings();
 	ERR_clear_error();
 
 	verbose_env = getenv("KBUILD_VERBOSE");
@@ -122,6 +122,10 @@ int main(int argc, char **argv)
 		fclose(f);
 		exit(0);
 	} else if (!strncmp(cert_src, "pkcs11:", 7)) {
+#ifdef OPENSSL_IS_AWSLC
+		ERR(1, "AWS-LC does not support extracting from PKCS#11");
+		exit(1);
+#else
 		ENGINE *e;
 		struct {
 			const char *cert_id;
@@ -144,6 +148,7 @@ int main(int argc, char **argv)
 		ENGINE_ctrl_cmd(e, "LOAD_CERT_CTRL", 0, &parms, NULL, 1);
 		ERR(!parms.cert, "Get X.509 from PKCS#11");
 		write_cert(parms.cert);
+	#endif
 	} else {
 		BIO *b;
 		X509 *x509;

```

Alternatively, either these patches could be upstreamed somehow to the Linux kernel, or the patches could live in a separate repository if that is preferred.

**Comment by samuel40791765** [comment]
Hi @theoparis, thanks for reaching out! We appreciate you trying out AWS-LC and we're also excited to patch/integrate into more projects for AWS-LC support. Unfortunately, AWS-LC is committed to having a stable API, but is not ABI stable. Systems cannot directly swap out OpenSSL with AWS-LC without recompiling. This makes us not suitable as a system library in a traditional Linux distribution.
Long-term we do want to support packaging AWS-LC for operating systems, but there's still a substantial amount of effort to be made before we can integrate AWS-LC into any linux distributions. You can check out https://github.com/aws/aws-lc/issues/1098 for more details regarding that.

**Comment by samuel40791765** [comment]
Closing in favor of https://github.com/aws/aws-lc/issues/1098 to package AWS-LC for operating systems. Feel free to reopen if there are any other questions, thank you!
